### PR TITLE
Feat/ccn3 2687/address from date

### PR DESCRIPTION
**Pull request by jenniferbrown94**
## Description

Nanny applicant address from date, PRs for ARC and Integration Adapter to follow

## Todo's before PR

- [ ] Branch has been rebased against develop
- [ ] Unit tests passed (`make test`)
- [ ] Selenium tests have passed
- [ ] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [ ] PR description describes the overall goals of PR commits
- [ ] Templates have been checked against the relevant user-story

## Migrations

- [ ] Null fields in models specifies default value
- [ ] You generated and applied migrations (`make migrate`)
- [ ] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [ ] Specified reviewers (even if it's yourself)
- [ ] Specified assignees (those who'll merge)
- [ ] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [ ] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

File: application/presentation/form_fields.py [hunk 1]
```diff
@@ -281,6 +285,8 @@ class CustomSplitDateField(forms.MultiValueField):
         'invalid': _('Please check the date of the course')
     }
 
+    TODAY = object()
```
Comment (rylandos) [inline application/presentation/form_fields.py]: Is this redundant?  The same variable is defined on line 390

File: application/presentation/personal_details/forms/home_address.py [hunk 2]
```diff
@@ -151,9 +188,30 @@ class HomeAddressLookupForm(NannyForm):
     error_summary_title = 'There was a problem'
     auto_replace_widgets = True
 
+    ERROR_MESSAGE_DATE_BLANK = 'Enter the full date, including the day, month and year'
```
Comment (rylandos) [inline application/presentation/personal_details/forms/home_address.py]: These are defined earlier in the file in a different class.  Is there any way to make them constants that are shared by both classes?  So we don't have to maintain them in two places

File: application/presentation/personal_details/forms/home_address.py [hunk 3]
```diff
@@ -142,6 +165,20 @@ def clean_postcode(self):
             raise forms.ValidationError('Please enter a valid postcode')
         return postcode
 
+    def clean_moved_in_date(self):
+        moved_in_date = self.cleaned_data['moved_in_date']
+        # get applicant's date of birth
```
Comment (rylandos) [inline application/presentation/personal_details/forms/home_address.py]: Date of birth?

File: application/presentation/personal_details/forms/home_address.py [hunk 4]
```diff
@@ -173,3 +231,17 @@ def __init__(self, *args, **kwargs):
 
     def clean_home_address(self):
         return int(self.cleaned_data['home_address'])
+
+    def clean_moved_in_date(self):
```
Comment (rylandos) [inline application/presentation/personal_details/forms/home_address.py]: Is there any chance that this method could be defined only once but used by both classes?

**Comment by jenniferbrown94** [comment]
Hi @rylandos , I can definitely move the error messages so they are only defined once, I'll see if I can do the same for the clean method 👍

**Pull request merged**